src/fmpy/fmi3.py

```diff
@@ -697,14 +697,29 @@ def getString(self, vr, nValues=None):
         self.fmi3GetString(self.component, vr, len(vr), value, nValues)
         return list(map(lambda b: b.decode('utf-8'), value))
 
-    def getBinary(self, vr, nValues=None):
+    def getBinary(self, vr: Iterable[int], nValues: int = None) -> Iterable[bytes]:
+
         if nValues is None:
             nValues = len(vr)
+
         vr = (fmi3ValueReference * len(vr))(*vr)
+
         value = (fmi3Binary * nValues)()
+
         size = (c_size_t * len(vr))()
+
         self.fmi3GetBinary(self.component, vr, len(vr), size, value, nValues)
-        return list(value)
+
+        values = []
+
+        for i, pointer in enumerate(value):
+            if pointer is None:
```

Comment:
An invalid pointer object here isn't None, it is something like `<fmpy.fmi3.LP_c_ubyte object at 0x7f001429e340>`

However, `bool(pointer)` will evaluate to False if the underlying pointer is invalid, and True if not.

So the check here should just be `if not pointer:`